internal/tls/tls_test.go

```diff
@@ -53,36 +55,138 @@ func TestNew(t *testing.T) {
+			args: args{
+				opts: []Option{
+					WithCert("./testdata/dummyServer.crt"),
+					WithKey("./testdata/dummyServer.key"),
+					WithCa("./testdata/dummyCa.pem"),
+				},
+			},
+			want: want{
+				want: func() *Config {
+					var err error
+					c := &credentials{
+						cfg: new(tls.Config),
+					}
+
+					c.cfg.Certificates = make([]tls.Certificate, 1)
+					c.cfg.Certificates[0], err = tls.LoadX509KeyPair("./testdata/dummyServer.crt", "./testdata/dummyServer.key")
+					if err != nil {
+						panic(err)
+					}
+
+					pool := x509.NewCertPool()
+					b, err := ioutil.ReadFile("./testdata/dummyCa.pem")
+					if err != nil {
+						panic(err)
+					}
+					if !pool.AppendCertsFromPEM(b) {
+						panic("faild to add cert")
+					}
+
+					c.cfg.ClientCAs = pool
+					c.cfg.ClientAuth = tls.RequireAndVerifyClientCert
+
+					c.cfg.BuildNameToCertificate()
+					return c.cfg
+				}(),
+			},
+			checkFunc: func(w want, c *tls.Config, err error) error {
+				if !errors.Is(err, w.err) {
+					return fmt.Errorf("got error = %v, wantErr %v", err, w.err)
+				}
+
+				if len(c.Certificates) != 1 && len(c.Certificates) != len(w.want.Certificates) {
+					return errors.New("Certificates length is wrong")
+				}
+				if got, want := string(w.want.Certificates[0].Certificate[0]), string(c.Certificates[0].Certificate[0]); want != got {
+					return errors.Errorf("Certificates[0] want: %v, but got: %v", want, got)
+				}
+
+				sl := len(c.ClientCAs.Subjects())
+				if sl == 0 {
+					return errors.New("subjects are empty")
+				}
+				if got, want := c.ClientCAs.Subjects()[sl-1], w.want.ClientCAs.Subjects()[0]; !reflect.DeepEqual(got, want) {
+					return errors.Errorf("ClientCAs.Subjects want: %v, got: %v", want, got)
+				}
+
+				if got, want := c.ClientAuth, w.want.ClientAuth; want != got {
+					return errors.Errorf("ClientAuth want: %v, but got: %v", want, got)
+				}
+				return nil
+			},
+		},
+		{
+			name: "returns nil and error when option is empty",
+			args: args{},
+			want: want{
+				err: errors.ErrTLSCertOrKeyNotFound,
+			},
+		},
+		{
+			name: "returns nil and error when cert path is empty",
+			args: args{
+				opts: []Option{
+					WithKey("./testdata/dummyServer.key"),
+				},
+			},
+			want: want{
+				err: errors.ErrTLSCertOrKeyNotFound,
+			},
+		},
+		{
+			name: "returns nil and error when key path is empty",
+			args: args{
+				opts: []Option{
+					WithCert("./testdata/dummyServer.crt"),
+				},
+			},
+			want: want{
+				err: errors.ErrTLSCertOrKeyNotFound,
+			},
+		},
+		{
+			name: "returns nil and error when contents of cert file is invalid",
+			args: args{
+				opts: []Option{
+					WithCert("./testdata/invalid.crt"),
+					WithKey("./testdata/dummyServer.key"),
+				},
+			},
+			want: want{
+				err: errors.New("tls: failed to find any PEM data in certificate input"),
+			},
+			checkFunc: func(w want, got *Config, err error) error {
+				if w.err.Error() != err.Error() {
```

Comment:
@hlts2 
for this checking, since the error is generated internally, so `errors.Is()` cannot check the error directly.
https://github.com/golang/go/blob/master/src/crypto/tls/tls.go#L296

Do you have any idea for it?

Reply:
I think that it will work normally if it is below...:thinking:

```

import stderrs "errors"
type want struct {
    err: stderrs.New(""tls: failed to find any PEM data in certificate input"")
}
```

Reply:
thx! it works.